src/type/schema.js

```diff
@@ -111,9 +111,9 @@ export class GraphQLSchema {
         '"allowedLegacyNames" must be Array if provided but got: ' +
           `${String(config.allowedLegacyNames)}.`,
       );
-      this.__allowedLegacyNames = config.allowedLegacyNames;
     }
 
+    this.__allowedLegacyNames = config.allowedLegacyNames;
```

Comment:
The `assumeValid` configuration is about whether or not to check the configuration for common mistakes, so regardless of that the allowed legacy names still need to be stored on the schema instance.

There’s no tests for `assumeValid` and the below configuration being set or not, so wasn’t sure what you’d like. Should I add a test for any configuration being set, regardless of `assumeValid`?

Reply:
Adding a test would be great!

Let me know if I understand this correctly - you could create a GraphQLSchema with *both* `__allowedLegacyNames` *and* `assumeValid`, then extend that schema with a legacy name (which might no longer be assumed invalid), and then expect validation to pass?

Reply:
Hmm, I now see that `validate` depends on the private `__validationErrors` property of a schema instance.

I’m now starting to think that maybe what Relay is doing isn’t entirely correct. First it creates a schema [with `assumeValid` set](https://github.com/facebook/relay/blob/181455c68f6870df93a1f1c2be0dcc0101399038/packages/relay-compiler/bin/RelayCompilerBin.js#L243), then after extending that schema a few times it expects to still be able to [validate that schema](https://github.com/facebook/relay/blob/181455c68f6870df93a1f1c2be0dcc0101399038/packages/relay-compiler/graphql-compiler/core/GraphQLValidator.js#L47), which appears to only really be possible because `extendSchema` does not copy over the `assumeValid` configuration.

Maybe `extendSchema` should _also_ copy over the `assumeValid` setting, but `validate` could accept an optional parameter to force performing the validations regardless of `__validationErrors` already being set?

Reply:
Not quite. extendSchema produces a new schema - while the provided schema may have been "valid" (either assumed or validated), the returned schema is different and may be invalid, so it must not copy over assumeValid or __validationErrors from the input. We would expect the extended schema to be validated by validateSchema().

Now, we may want to provide a similar assumeValid option for extendSchema, but that would be a separate thing.

validate checks __validationErrors first as a form of memoization. Since schema are considered immutable, we do not want to wastefully validate the same schema multiple times.

Reply:
Relay first builds a schema with the assumeValid set, because it expects that schema has been provided from a valid GraphQL server - whereas any client side extensions are within Relay's responsibility to validate

Reply:
Also, the `GraphQLValidator` second link is validating the graphql operations and fragments found within relay product code, not the schema

Reply:
Ok, I see, that makes sense. So to answer your initial question, when an original schema has `__allowedLegacyNames` then extending that schema should still allow those original legacy names too, regardless from whether or not they were validated on the original schema.

Reply:
That `validate` call in `GraphQLValidator` is definitely still leading to an error about the legacy name in the schema . Maybe it’s just a side-effect, though, I’ll double-check that in the morrow.

Reply:
Probably a side effect - the document validator first asserts that a valid schema if being used to validate a document. If the schema wasn't validated before that point, it will throw